State.cpp

```diff
@@ -0,0 +1,248 @@
+#include "BreathData.h"
+#include "State.h"
+#include "Constants.h"
+#include "Display.h"
+#include "Flow.h"
+#include "Pressure.h"
+#include "Valve.h"
+#include "ProportionalValve.h"
+#include "AlarmManager.h"
+#include "State.h"
+
+// Initialize singleton instances
+OffState* OffState::instance = 0;
+InsStateVC* InsStateVC::instance = 0;
+InsHoldStateVC* InsHoldStateVC::instance = 0;
+ExpStateVC* ExpStateVC::instance = 0;
+PeepStateVC* PeepStateVC::instance = 0;
+ExpHoldStateVC* ExpHoldStateVC::instance = 0;
+
+BreathData* State::breath;
```

Comment:
Instead of being a pointer, why not just have a `BreathData` object?

Reply:
I was wanting to have the BreathData object created elsewhere and passed in here, and I couldn't do that with a plain object.  I think this is another example of me trying to use Java idioms that aren't appropriate.  I'll take your advice.